Updates for new AliGenerators release

qgp opened the pull request:
@dberzano, @ktf, @rbertens:

Do NOT merge yet!

This PR is to bump the versions of generators included in AliGenerators. Furthermore, it adds Rivet and Rivet-hi. As of now, the recipes require building with ```--defaults pwgmmtest``` as not all source code is pushed to alisw yet.

We would like to ask for a test build to alice-nightlies (with pwgmmtest) as done for validation before.

Cheers,
   Jochen

dberzano commented:
I'll keep you posted!

dberzano commented:
@qgp I see a relevant change here: ROOT 6 as a base. This will make impossible to load at the same time AliPhysics and AliGenerators (but maybe it's not required). I believe this is intended though, right?

qgp commented:
@dberzano: Good catch! Indeed, basing this on ROOT 6 was done intentionally to see if things are working with the future ROOT. And I think, it is what we should test and deploy in the future. I'd rather consider a ROOT5-based build a legacy build now.

But surely open for discussion ...

dberzano commented on defaults-pwgmmtest.sh:
```diff
@@ -5,15 +5,57 @@ env:
+    version: "%(commit_hash)s_PWGMMTEST"
+    tag: v5-09-22-01
+# now for actual generators
+  agile:
```
Can you add AliGenerators too? We need the `_PWGMMTEST` suffix for that package (see alisw/ali-bot#540).

qgp replied:
Sure, done.

dberzano commented:
The plan is to start the migration to ROOT 6 after QM. The AliDPG group for instance has started updating all macros already. We'll make a clear stance on this during the Offline Week. On my side, _if using ROOT 6 for the PWGMM conveners is fine_, this PR is OK, but the incompatibility with AliPhysics still stands.

dberzano commented:
Note to self: while [the test build is finally running](https://alijenkins.cern.ch/job/build-any-ib/6644/consoleFull), a small change in the ROOT recipe was needed (added to this PR) and the test build is running on SLC6 (not SLC5).

qgp commented:
Little heads up on the current status for @dberzano and @rbertens:
done:
- alidist/herwig.sh: added missing dependency on GMP
- Rivet patched to fix data retrieval from HepData
- LHAPDF: updated to 6.2.1 and split into lhapdf and lhapdf-pdfsets

to be done:
- debug rivet plotting on lxplus (maybe trivial, maybe not, doesn't work right with the current test version but haven't really looked into it yet)

Cheers,
   Jochen

dberzano commented:
Do you want me to rebuild?

qgp commented:
Hi @dberzano, sorry for the long silence. We still need an update of YODA (required by the updated Rivet version) and there's a problem with the separation of LHAPDF and the PDF sets. I'm working on these items, as soon as I have a working version, I'll update the PR. And, then, it would be good if we could have a rebuild of PWGMMTEST.

Cheers,
   Jochen

qgp commented:
Hi @dberzano, 

finally, an update. There still was an issue with Herwig after having split the pdfsets from LHAPDF. In fact, I need to explicitly set the environment variable LHAPDF_DATA_PATH in the build recipe (herwig.sh). Setting in the yaml header using env/prepend_path does not seem to work. Could you have a look at this recipe?

As soon as we have clarified this issue with Herwig, I'd like to ask for a new test build.

Cheers,
   Jochen

rbertens commented:
@dberzano and @qgp : I hate to sound pushy, but a number of analyses are depending on the updates in the PR. Would be great if this could be resolved quickly!

dberzano commented:
Dear all, I have just rebased & pushed a cleanup that addresses the HERWIG issue. `env:` is used to set the runtime environment for a package, not the build environment, therefore the right thing to do is to manually `export` the required variable inside the recipe's body.

qgp commented:
Hi @dberzano,

thanks for the clarification and the update. Then we are ready to go for the test build.

Cheers,
   Jochen

dberzano commented:
Build works now, it should be fine for testing. You can check the [Jenkins log](https://alijenkins.cern.ch/job/build-any-ib/6694/) if you want. When done, the usual email will be received. You can then start the testing and confirm that everything works as expected!

Just to be 100% sure of what we are doing here:

* This set of `AliGenerators` will be based on ROOT 6
* Currently, AliEn is tolerant when loading at the same time incompatible packages: it sends warnings that users ignore
* We are turning this warnings into **hard errors** in order to prevent users from accidentally loading ROOT 5 and ROOT 6 implicitly (we also build them on different platforms, so this would be a recipe for disaster)

dberzano commented:
OK it appears the publishing on lxplus worked, now it's time for @rbertens and @qgp for running the actual tests. Note this is still a WIP until we get confirmation it works!

qgp commented:
Thanks @dberzano, first tests look good, but more tests to come. We'll confirm when we are happy.

For lhapdf-pdfsets the PWGMMTEST suffix was missing, just committed a fix (only relevant when using defaults-pwgmmtest.sh).

Propagation of code from alipwgmm to alisw started to allow for official build (not using defaults-pwgmmtest.sh)

dberzano commented:
[New build started.](https://alijenkins.cern.ch/job/build-any-ib/6696/console)

dberzano commented:
Before merging, I am testing on Ubuntu whether we have problems with Herwig builds (as @qgp reported privately). It seems to me that Herwig might have random build problems... no problem at all on our centralized builds though. I'll report ASAP.

dberzano commented:
I can cleanly compile on Ubuntu 16.04 (using the native compiler).

dberzano commented:
I have factored out #1091. I prefer to merge that one first, and then this one.

alibuild commented:
Error while checking build/O2/alidist for 2134c7376ec3d41aad816a89214076105b5d8b15:
```
sw/BUILD/O2-latest/log
[63607]: [00:50:13][STATE] Entering INITIALIZING DEVICE state
[63606]: port-range-max                                  = 32000                                                                                               <int>            [default] 
[63606]: port-range-min                                  = 22000                                                                                               <int>            [default] 
[63606]: rate                                            = 0                                                                                                   <float>          [default] 
[63606]: session                                         = default                                                                                             <string>         [default] 
[63606]: severity                                        = debug                                                                                               <string>         [default] 
[63606]: shm-monitor                                     = 0                                                                                                   <bool>           [default] 
[63606]: shm-segment-size                                = 2000000000                                                                                          <std::size_t>    [default] 
[63606]: transport                                       = zeromq                                                                                              <string>         [default] 
[63606]: verbosity                                       = medium                                                                                              <string>         [default] 
[63606]: 
[63606]: [00:50:13][STATE] Entering INITIALIZING DEVICE state
[63606]: 
[63606]: [00:50:13][STATE] Entering DEVICE READY state
[63606]: [00:50:13][STATE] Entering INITIALIZING TASK state
[63606]: [00:50:13][STATE] Entering READY state
[63606]: [00:50:13][STATE] Entering RUNNING state
[63606]: [00:50:13][INFO] DEVICE: Running...
[63606]: [00:50:13][INFO] foo[0]: in: -nan (-nan MB) out: -nan (-nan MB)
[63606]: [00:50:13][INFO] from_producer_to_consumer[0]: in: -nan (-nan MB) out: -nan (-nan MB)
[INFO] Quitting
[INFO] QUIT_REQUESTED

[63607]: [00:50:13][INFO] Received device shutdown request (signal 15).
[63607]: [00:50:13][INFO] Waiting for graceful device shutdown. Hit Ctrl-C again to abort immediately.
[63607]: [00:50:13][STATE] Entering DEVICE READY state
[63607]: [00:50:13][STATE] Entering INITIALIZING TASK state
[63607]: [00:50:13][STATE] Entering READY state
[63607]: [00:50:13][STATE] Entering RUNNING state
[63607]: [00:50:13][INFO] DEVICE: Running...
[63607]: [00:50:13][STATE] Entering READY state
[63606]: [00:50:13][INFO] Received device shutdown request (signal 15).
[63606]: [00:50:13][INFO] Waiting for graceful device shutdown. Hit Ctrl-C again to abort immediately.
[63606]: [00:50:14][STATE] Entering READY state
[63606]: [00:50:14][STATE] Entering RESETTING TASK state
[63606]: [00:50:14][STATE] Entering DEVICE READY state
[63606]: [00:50:14][STATE] Entering RESETTING DEVICE state
[63606]: [00:50:14][STATE] Entering IDLE state
[63606]: [00:50:14][STATE] Entering EXITING state
[63606]: [00:50:14][STATE] Exiting FairMQ state machine
[63606]: [00:50:14][INFO] Process 63606 is exiting.


99% tests passed, 1 tests failed out of 176

Total Test time (real) =  71.98 sec

The following tests FAILED:
	156 - test_Framework_test_WorkflowOptions (Timeout)
```
Full log [here](https://ali-ci.cern.ch/repo/logs/alisw/alidist/1036/2134c7376ec3d41aad816a89214076105b5d8b15/build_O2_alidist/fullLog.txt).

alibuild commented:
Error while checking build/O2/alidist for 1e045db25fc64c5e2f7cc9260eb912a9a6f57563:
```
sw/BUILD/O2-latest/log
[4997]: port-range-max                                  = 32000                                                                                               <int>            [default] 
[4997]: port-range-min                                  = 22000                                                                                               <int>            [default] 
[4999]: [16:40:29][STATE] Entering INITIALIZING DEVICE state
[4997]: rate                                            = 0                                                                                                   <float>          [default] 
[4997]: session                                         = default                                                                                             <string>         [default] 
[4997]: severity                                        = debug                                                                                               <string>         [default] 
[4997]: shm-monitor                                     = 0                                                                                                   <bool>           [default] 
[4997]: shm-segment-size                                = 2000000000                                                                                          <std::size_t>    [default] 
[4997]: transport                                       = zeromq                                                                                              <string>         [default] 
[4997]: verbosity                                       = medium                                                                                              <string>         [default] 
[4997]: 
[4997]: [16:40:29][STATE] Entering INITIALIZING DEVICE state
[4997]: 
[4997]: [16:40:29][STATE] Entering DEVICE READY state
[4997]: [16:40:29][STATE] Entering INITIALIZING TASK state
[4997]: [16:40:29][STATE] Entering READY state
[4997]: [16:40:29][STATE] Entering RUNNING state
[4997]: [16:40:29][INFO] DEVICE: Running...
[4997]: [16:40:29][INFO] foo[0]: in: -nan (-nan MB) out: -nan (-nan MB)
[4997]: [16:40:29][INFO] from_producer_to_consumer[0]: in: -nan (-nan MB) out: -nan (-nan MB)
[INFO] Quitting
[INFO] QUIT_REQUESTED

[4999]: [16:40:29][INFO] Received device shutdown request (signal 15).
[4999]: [16:40:29][INFO] Waiting for graceful device shutdown. Hit Ctrl-C again to abort immediately.
[4997]: [16:40:29][INFO] Received device shutdown request (signal 15).
[4997]: [16:40:29][INFO] Waiting for graceful device shutdown. Hit Ctrl-C again to abort immediately.
[4999]: [16:40:29][STATE] Entering DEVICE READY state
[4999]: [16:40:29][STATE] Entering INITIALIZING TASK state
[4999]: [16:40:29][STATE] Entering READY state
[4999]: [16:40:29][STATE] Entering RUNNING state
[4999]: [16:40:29][INFO] DEVICE: Running...
[4999]: [16:40:29][STATE] Entering READY state
[4997]: [16:40:30][STATE] Entering READY state
[4997]: [16:40:30][STATE] Entering RESETTING TASK state
[4997]: [16:40:30][STATE] Entering DEVICE READY state
[4997]: [16:40:30][STATE] Entering RESETTING DEVICE state
[4997]: [16:40:30][STATE] Entering IDLE state
[4997]: [16:40:30][STATE] Entering EXITING state
[4997]: [16:40:30][STATE] Exiting FairMQ state machine
[4997]: [16:40:30][INFO] Process 4997 is exiting.


99% tests passed, 1 tests failed out of 176

Total Test time (real) =  68.24 sec

The following tests FAILED:
	156 - test_Framework_test_WorkflowOptions (Timeout)
```
Full log [here](https://ali-ci.cern.ch/repo/logs/alisw/alidist/1036/1e045db25fc64c5e2f7cc9260eb912a9a6f57563/build_O2_alidist/fullLog.txt).
